src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -253,4 +253,17 @@ public static long transferVectors(long vectorsPointer, float[][] trainingData)
     public static void freeVectors(long vectorsPointer) {
         FaissService.freeVectors(vectorsPointer);
     }
+
+    /**
+     * Range search index for a given query vector
+     *
+     * @param indexPointer pointer to index in memory
+     * @param queryVector vector to be used for query
+     * @param radius search within radius threshold
+     * @return KNNQueryResult array of neighbors within radius
+     */
+    public static KNNQueryResult[] radiusQueryIndex(long indexPointer, float[] queryVector, float radius) {
```

Comment:
I meant to remove KNNEngine from FaissService, not JNIService. Keep KNNEngine in arg for this.

Reply:
ack